### PR TITLE
CHE-4865: Correctly detect installed java version

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Correctly detect installed java version

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/4865

#### Changelog
Correctly detect installed java version

#### Release Notes
not required

#### Docs PR
not required

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
Tested with `codenvy/jdk7`, `codenvy/codenvy` and `eclipse/ubuntu_jdk8` images

File: wsagent/agent/src/main/resources/org.eclipse.che.ws-agent.script.sh [hunk 1]
```diff
@@ -52,8 +52,8 @@ INSTALL_JDK=false
 command -v ${JAVA_HOME}/bin/java >/dev/null 2>&1 || {
     INSTALL_JDK=true;
 } && {
-    java_version=$(${JAVA_HOME}/bin/java -version 2>&1 | sed 's/.* version "\\(.*\\)\\.\\(.*\\)\\..*"/\\1\\2/; 1q')
-    if [ ! -z "${java_version##*[!0-9]*}" ] && [ "${java_version}" -lt "18" ]; then
+    java_version=$(${JAVA_HOME}/bin/java -version 2>&1 | grep version  | awk '{print $NF}' | sed 's/"//g' | cut -d '.' -f2)
```
Comment (garagatyi) [inline wsagent/agent/src/main/resources/org.eclipse.che.ws-agent.script.sh:55]: There is no checks that awk is present in container. awk is used in centos check, so I supose in works on centos but how to be sure it works in other images?.

Comment (ghost) [reply wsagent/agent/src/main/resources/org.eclipse.che.ws-agent.script.sh:55]: Tried ubuntu, debian and alpine.

Comment (garagatyi) [reply wsagent/agent/src/main/resources/org.eclipse.che.ws-agent.script.sh:55]: What about user's custom images? Have we declared that awk is mandatory in dev machine?

Comment (ghost) [reply wsagent/agent/src/main/resources/org.eclipse.che.ws-agent.script.sh:55]: Library images for alpine, busybox, ubuntu, debian, centos, fedora and rhel have awk.

Also, in all other agents we use awk - so far so good. So, if we are to check if awk is available, then we should do it across all agent scripts. However, having a custom image without awk is quite uncommon I think (maybe some super stripped images)

Comment (garagatyi) [reply wsagent/agent/src/main/resources/org.eclipse.che.ws-agent.script.sh:55]: ok then

**Review by garagatyi — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Pull request merged**